### DISCUSSION TITLE
Bubble Menu with Livewire/AlpineJS

### DISCUSSION THREAD

**Discussion by amnesca** [ueberdosis/tiptap#5014] [General Questions]
Hi, 
I set up Tiptap using the [Laravel Livewire](https://tiptap.dev/docs/editor/installation/php#laravel-livewire) installation and initially couldn't seem to get the [Bubble Menu](https://tiptap.dev/docs/editor/api/extensions/bubble-menu) extension working. It didn't throw any errors but the menu would just appear static and not display when selecting text in the editor, as it should. My guess was that Alpine hadn't finished rendering and wasn't connecting to the menu element, so I added a [nextTick](https://alpinejs.dev/directives/init#next-tick) property to the init directive for the editor reference and it's now working properly (see snippet below). However, as this wasn't in the official Tiptap documentation, I just wanted to confirm that this is indeed the correct method to connect extensions with Livewire/AlpineJS? 
Thank you.

```html
<div
  x-data="setupEditor(
    $wire.entangle('{{ $attributes->wire('model')->value() }}').defer,
    '{{ $attributes['user'] }}'
  )"
  x-init="$nextTick(() => { init($refs.editorReference) })"
  wire:ignore
  {{ $attributes->whereDoesntStartWith('wire:model') }}
>
	<template x-if="isLoaded()">
		<div class="editor-bubble-menu>
			<button @click="toggleBold()" :class="{ 'is-active' : isActive('bold', updatedAt) }">
		            BOLD
		        </button>
		</div>
	</template>
	
	<div x-ref="editorReference"></div>
</div>

```

**Comment by seabasss** [comment]
+1 on this. For me the bubble menu shows up if I remove `<template x-if="isLoaded()">` but then I get initialization errors and problems with the active state. Any ideas? Thanks!

**Comment by seabasss** [comment]
@amnesca would you be open to sharing your js code for this as well? I see that you're using `updatedAt` that is not in the original code for example. Thank you!

**Reply by amnesca** [reply]
Hey @seabasss, I integrated the JS from the [Alpine setup](https://tiptap.dev/docs/editor/installation/alpine), which has an example of adding a fixed menu.

**Reply by seabasss** [reply]
Thanks! That’s what I ended up doing too, but I only got the bubble menu to work after adding your next tick suggestion and using x-init, which the alpine example didn’t do. It still flickers if I replace the model that’s being edited, but I’ll find a way to get around it.

If you’re keen on sharing your js that would be cool, but no worries if not. Just interesting to compare.

**Comment by seabasss** [comment]
Also, any intel on how to deal with throttling? `.defer` works for when you submit, I can add something like this for `onBlur`, but what's the best approach for throttling?

```
onBlur({ editor }) {
    _this.$wire.set(property, editor.getHTML())
},
```
